### ISSUE TITLE
`NewSDKConfig()` hardcoded to use very old credentials file

### ISSUE THREAD

**Issue by nkubala** [golang/oauth2#382]
At some point in the past, the Cloud SDK switched from using a simple `<config_dir>/credentials` file to store authentication information, to a SQLite db file `<config_dir>/credentials.db`. The only information I can find about this at all is this [SO post](https://stackoverflow.com/questions/48088841/gcloud-auth-login-saves-to-legacy-credentials-folder). (not sure if this is actually formally documented somewhere)

However, `NewSDKConfig()` is still hardcoded to use this old creds file: https://github.com/golang/oauth2/blob/9f3314589c9a9136388751d9adae6b0ed400978a/google/sdk.go#L59

meaning that since most users have upgraded their gcloud versions since this change to the SDK creds, `NewSDKConfig()` will fail the majority of the time.

If ADCs are generated by the user this flow won't be used, but for those who don't have ADCs generated and are only authenticated through `gcloud auth login`, we should still be able to generate an authenticated client.

**Comment by bradfitz** [comment]
Ugh, so we have to shell out to gcloud to get at creds now?

/cc @jba @jadekler @broady

**Comment by broady** [comment]
Yes, that's the only way the gcloud team wants us accessing gcloud CLI credentials.

We should probably deprecate `NewSDKConfig`, it was hastily added to this package and didn't follow the contract set out by the gcloud CLI.

**Comment by nkubala** [comment]
anyone know if there's a plan to make this less painful from the client's perspective? right now running `gcloud auth login` and then doing a `gcloud builds submit` works fine, but there's no way to accomplish this same behavior through client libraries because these credentials can't be retrieved. we don't even have the option to create the ADCs through the client; we have to run `gcloud auth application-default login` to generate ADCs, which is less than ideal. do you guys know if the gcloud team has plans to expose the "active user's" credentials?
